modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -561,7 +561,12 @@ func (h *Handler) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 		} else if commonScheme == "https" {
 			return d.Errf("upstreams are configured for HTTPS but transport module does not support TLS: %T", transport)
 		}
-		if !reflect.DeepEqual(transport, reflect.New(reflect.TypeOf(transport).Elem()).Interface()) {
+
+		// if the transport is the same as the default,
+		// i.e. http and with no transport options, then
+		// we'll omit setting TransportRaw to avoid redundant
+		// JSON configuration.
+		if transportModuleName != "http" || !reflect.DeepEqual(transport, reflect.New(reflect.TypeOf(transport).Elem()).Interface()) {
```

Comment:
This can be much simpler, comment included:

```suggestion
		// no need to encode empty default transport
		if !reflect.DeepEqual(transport, new(HTTPTransport)) {
```